### ISSUE TITLE
Line indented incorrectly false positive when using match-expression inside switch case

### ISSUE THREAD

**Issue by i350** [squizlabs/PHP_CodeSniffer#3530]
**Describe the bug**
When I uses `match-expression` inside a `switch-case`, I get non logical error

**Code sample**
```php
    switch ($appEnv) {
        case 'production':
            return ($foo == 'bar') ? 'prod' : 'production';
        case 'staging':
            return match ($foo) {
                'bar' => 'custom_1',
                'baz' => 'custom_2',
                default => 'staging'
            };
        case 'testing':
            return 'testing';
        default:
            return 'default';
    }
```
I get error starting from the line "case 'testing': .....'
```
Error: Line indented incorrectly; expected 12 spaces, found 8
```
This require me to justify the "case 'testing':" with its body and "default:" case with its body to closing curly brace of match expression! 
Like this:
```php
    switch ($appEnv) {
        case 'production':
            return ($foo == 'bar') ? 'prod' : 'production';
        case 'staging':
            return match ($foo) {
                'bar' => 'custom_1',
                'baz' => 'custom_2',
                default => 'staging'
            };
            case 'testing':
                return 'testing';
            default:
                return 'default';
    }
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
PHPCS output here
```

**Expected behavior**

**Versions (please complete the following information):**
 - OS: Linux 050918c5cfc1 5.11.0-43-generic ~ 20.04.2-Ubuntu SMP x86_64
 - PHP: 8.0.13
 - PHPCS: 3.6.1 stable
 - Standard: PSR2

**Comment by gsherwood** [comment]
This problem was caused by the match expression being returned instead of assigned to a variable and then returned. The return here is special as it is the closer for the case statement, so everything got confused. There was already code to handle this for anon classes and closures, so I treat match in the same way.

Thanks for reporting this. The fix will be in 3.7.0.
